kotlin/src/test/kotlin/org/axonframework/extensions/kotlin/serialization/KotlinSerializerTest.kt

```diff
@@ -0,0 +1,95 @@
+package org.axonframework.extensions.kotlin.serialization
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import org.axonframework.serialization.AnnotationRevisionResolver
+import org.axonframework.serialization.ChainingConverter
+import org.axonframework.serialization.SerializedType
+import org.axonframework.serialization.SimpleSerializedObject
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class KotlinSerializerTest {
+
```

Comment:
I am missing test cases that validate if the `KotlinSerializer` can be used for Axon objects, like the `Message`, `TrackingToken`, and `SagaEntry`.

Assuming that's because those objects aren't annotated with the `@Serializable` annotation.
However, the `Serializer` in Axon must de-/serialize those objects for it to be usable.

Reply:
Indeed. See https://github.com/AxonFramework/extension-kotlin/pull/124/files#r645048334 for the comment about serializing internal classes.

I added one test case for the `ConfigToken`, and every internal class that should be serializable can be added in that way, if there is a serializer.

Reply:
Gotcha. To make this `Serializer` implementation workable for Axon Framework entirely, these tests will be a requirement too.

Making these can wait until we've concluded the conversation on the `Serializer` implementation though.
On the subject of using the `@Serializable` annotation on Axon components, and if that even works for Java classes, that is.